monai/_extensions/loader.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import signal
+from glob import glob
+from os import makedirs, path, remove
+from shutil import rmtree
+from sys import platform
+from time import sleep
+
+from torch import cuda
+from torch.utils.cpp_extension import load
+
+dir_path = path.dirname(path.realpath(__file__))
+lock_max_wait = 30
+lock_check_interval = 5
```

Comment:
could these `dir_path`, `lock_max_wait` be arguments of `load_module(...)`, so that `load_module` could be a generic API for MONAI users to compile their own extensions anywhere on PYTHONPATH?

Reply:
I've updated the timeout code and added the timeout limit as an argument to the `load_module` function, This way it can account for extentions with longer compile times where needd. I dont think this function would be useful to MONAI users as its just a wrapper for torch's `cpp_extention.load` which reduces the functionality.

Reply:
thanks, the `build_dir` is currently hard-coded as a subfolder of the `module_dir`, is it really needed? [`torch.utils.cpp_extension.load`](https://pytorch.org/docs/stable/cpp_extension.html#torch.utils.cpp_extension.load) has an option of setting the `TORCH_EXTENSIONS_DIR` environment variable. Shall we preserve that behaviour?